File: 00_playground/01_short_circuit/shortcircuit.py
def main():
    print("Doing 'and' test")
    if returnFalse() and returnTrue():
        print("'And' test: This should never appear!")

    print("Done with 'and' test")

    print("\nDoing 'or' test")
    if returnTrue() or returnFalse():
        print("'Or' test: This should always appear!")

    print("Done with 'or' test")


def returnTrue():
    print("Returning true!")
    return True


def returnFalse():
    print("Returning false!")
    return False


main()

File: 00_playground/01_short_circuit/shortcircuit.c
/*
        compile: gcc shortcircuit.c -o shortcircuit
        run:    ./shortcircuit
*/

#include <stdio.h>

int returnTrue(void);
int returnFalse(void);

int main(void) {
        // testing short circuit with and
        printf("Doing 'and' test\n");
        if (returnFalse() && returnTrue()) {
                printf("'And' test: this should never appear!\n");
        }
        printf("Done with 'and' test\n");

        // testing short circuit with or
        printf("\nDoing 'or' test\n");
        if (returnTrue() || returnFalse()) {
                printf("'Or' test: this should always appear!\n");
        }
        printf("Done with 'or' test\n");
}

int returnTrue(void) {
        printf("Returning true!\n");
        return 1;        // no bools in C, so return 1 for "true"
}

int returnFalse(void) {
        printf("Returning false!\n");
        return 0;        // no bools in C, so return 0 for "false"
}
